source/extensions/filters/http/common/jwks_fetcher.cc

```diff
@@ -107,18 +148,69 @@ class JwksFetcherImpl : public JwksFetcher,
   bool complete_{};
   JwksFetcher::JwksReceiver* receiver_{};
   const envoy::config::core::v3::HttpUri* uri_{};
+
   Http::AsyncClient::Request* request_{};
 
+  Tracing::Span* parent_span_{};
+
+  const envoy::extensions::filters::http::jwt_authn::v3::RemoteJwks& remote_jwks_;
+
+  Envoy::BackOffStrategyPtr backoff_strategy_;
+
+  uint32_t num_retries_;
+
+  uint32_t retries_remaining_;
+
+  Envoy::Random::RandomGeneratorImpl random_;
+
   void reset() {
     request_ = nullptr;
     receiver_ = nullptr;
     uri_ = nullptr;
+    parent_span_ = nullptr;
+
+    // truncated backoff back to 0 retries attempted.
+    retries_remaining_ = num_retries_;
+
+    if (backoff_strategy_) {
+      // backoff strategy : back to initial (small) delay
+      backoff_strategy_->reset();
+    }
+  }
+
+  void retryFetch() {
+
+    // cant' fetch() if receiver isn't null... ( for example after a reset() )
+    auto* receiver = receiver_;
+    receiver_ = nullptr;
+
+    if (backoff_strategy_) {
+      if (retries_remaining_-- > 0) {
+
+        auto retry_ms = std::chrono::milliseconds(backoff_strategy_->nextBackOffMs());
+
+        ENVOY_LOG(warn, "retrying after {} milliseconds backoff", retry_ms.count());
+
+        absl::SleepFor(absl::FromChrono(retry_ms));
```

Comment:
No,  you could not put sleep here.  You have to pass in "dispatcher" and use it to create a timer

Reply:
done.